app/src/ui/diff/text-diff.tsx

```diff
@@ -1004,6 +1007,11 @@ export class TextDiff extends React.Component<ITextDiffProps, ITextDiffState> {
     if (batchedOps.length > 0) {
       cm.operation(() => batchedOps.forEach(x => x()))
     }
+
+    // We have to set the size of the gutter, and because it might have changed we should refresh to fit the editor.
+    const gutterElement = document.getElementsByClassName('diff-gutter')[0]
+    gutterElement.setAttribute('style', `width: ${diffSize * 2}px;`)
```

Comment:
Instead of changing the style directly, hardcoding a width, I think it'd be cleaner (and more consistent with the rest of the codebase) to have a set of "diff gutter width" classes depending on the size we need. For example:
```css
--diff-gutter-width-per-char: 15px;

// Width for 1-digit line numbers
.diff-gutter-width-1:
  width: calc(var(--diff-gutter-width-per-char) + 5);

// Width for 2-digit line numbers
.diff-gutter-width-2:
  width: calc(var(--diff-gutter-width-per-char) * 2);

// Width for 3-digit line numbers
.diff-gutter-width-2:
  width: calc(var(--diff-gutter-width-per-char) * 3);

...
```

Then, you apply a class to the diff gutter depending on the number of digits. Does that make sense?

Reply:
Also /cc @tidy-dev to get her input, I understand the convenience of just hardcoding the exact width depending on the number of digits, but it feels "wrong" 😳

Reply:
Agree using css is generally optimal when applying styles as you are letting the css control the view. Doing so would necessitate having an upper limit, because you are limited to the number of classes you make. IMHO.. the diff viewer is a unique case because it inherently must calculates things dynamically for the view. Thus, it doesn't feel that wrong to me. I think due to the nature of how the lines are rendered we are stuck with a non-optimal solution of calculating it where in an ideal world these things would be in columns that we could just add padding and it would adjust itself to the widest contents. 

Long story short: My 2 cents is I feel either solution would be acceptable.

Reply:
I don't know if my input makes a difference, but I will say that hard coding css values would work since the upper limit is ignored. The difference is the dynamic version will allow the large numbers to work with the amount of space for text possibly looking wonky, but the other will just truncate the numbers since it'll only get so large.

I'd personally prefer dynamic.

Reply:
👍  let's go ahead with the dynamic approach then!